lib/samples/navigate_route/navigate_route.dart

```diff
@@ -0,0 +1,424 @@
+    await _simulatedLocationDataSource.start();
+
+    // Create a RouteTrackerLocationDataSource.
+    final routeTrackerLocationSource = RouteTrackerLocationDataSource(
+      routeTracker: _routeTracker,
+      locationDataSource: _simulatedLocationDataSource,
+    );
+
+    // Set the location display data source.
+    _mapViewController.locationDisplay.dataSource = routeTrackerLocationSource;
+
+    // Set the auto-pan mode for the location display.
+    _mapViewController.locationDisplay.autoPanMode =
+        LocationDisplayAutoPanMode.navigation;
+
+    // Start the location display.
+    _mapViewController.locationDisplay.start();
+  }
+
+  void _updateRouteGraphics(TrackingStatus status) {
+    // Update the graphics for the route traveled and ahead.
+    setState(() {
+      _routeTravelledGraphic.geometry = status.routeProgress.traversedGeometry;
+      _routeAheadGraphic.geometry = status.routeProgress.remainingGeometry;
+    });
+  }
+
+  String formatDuration(Duration duration) {
+    final hours = duration.inHours;
+    final minutes = duration.inMinutes.remainder(60);
+    final seconds = duration.inSeconds.remainder(60);
+
+    if (hours > 0) {
+      return '$hours Hours, $minutes Minutes, and $seconds Seconds';
+    } else if (minutes > 0) {
+      return '$minutes Minutes and $seconds Seconds';
+    } else {
+      return '$seconds Seconds';
+    }
+  }
+
+// Usage
+  void _updateStatusText(TrackingStatus status) {
+    final remainingTime = Duration(seconds: status.routeProgress.remainingTime.round());
+    final formattedTime = formatDuration(remainingTime);
+    _statusTextNotifier.value = '''
+  Distance remaining: ${status.routeProgress.remainingDistance.displayText} ${status.routeProgress.remainingDistance.displayTextUnits.abbreviation}
+  Time remaining: $formattedTime
+  Next direction: ${_directionsList[status.currentManeuverIndex + 1].directionText}
+  ''';
+  }
+
+
+  void recenter() {
+    _mapViewController.locationDisplay.autoPanMode = LocationDisplayAutoPanMode.navigation;
+  }
+
+  Future<void> resetNavigation() async {
+    // Reset the navigation state.
+    setState(() => _ready = false);
+
+    // Stop the location display.
+    _mapViewController.locationDisplay.stop();
+
+    // Solve the route again.
+    await solveRoute();
+
+    // Clear the existing graphics.
+    _routeGraphicsOverlay.graphics.clear();
+
+    // Add the new route graphics to the overlay.
+    _routeGraphicsOverlay.graphics.add(_routeAheadGraphic);
+    _routeGraphicsOverlay.graphics.add(_routeTravelledGraphic);
+
+    // Stop any ongoing speech.
+    await _flutterTts.stop();
+
+    // Reset the status text.
+    _statusTextNotifier.value = 'Directions are shown here.';
+
+    // Set the viewpoint to encompass the route geometry.
+    setInitialViewpoint();
+
+    // Update the state to reflect the reset.
+    setState(() => _ready = true);
+  }
+
+  void speechEngineInUse() {
+    _routeTracker.setSpeechEngineReady(() => false);
```

Comment:
This feels unnecessarily complicated, to keep on changing the callback. All we need is a member variable `var _speechEngineReady = true` and just flip that variable back and forth. We only need to call "setSpeechEngineReady" once, probably up in "_createAndConfigureRouteTracker", and it just needs to be `_routeTracker.setSpeechEngineReady(() => _speechEngineReady);`.